OpenRA.Mods.Common/Traits/SupportPowers/GrantUpgradePower.cs

```diff
@@ -26,7 +26,7 @@ class GrantUpgradePowerInfo : SupportPowerInfo
 		[Desc("Duration of the upgrade (in ticks). Set to 0 for a permanent upgrade.")]
 		public readonly int Duration = 0;
 
-		[Desc("Cells")]
+		[Desc("Cells - affects whole cells only")]
 		public readonly int Range = 1;
```

Comment:
Its behaviour also includes a footprint like overlay for selection which would also have to be replaced. Players may be surprised when units that would be affected are not anymore.